app/src/main/java/org/greenstand/android/TreeTracker/fragments/MapsFragment.kt

```diff
@@ -140,7 +141,9 @@ class MapsFragment : androidx.fragment.app.Fragment(), OnClickListener, OnMapRea
 
     override fun onClick(v: View) {
         v.vibrate()
-
+        // Disable the addTreeButton below to avoid triggering the onClick listener
+        // more than one once
+        (v.findViewById(R.id.addTreeButton) as Button)?.isEnabled = false
```

Comment:
This disables the button right after it's clicked. What happens if we want to add another tree? Will the button still be disabled?
If that's the case, what about adding a delay between clicks: https://stackoverflow.com/questions/16534369/avoid-button-multiple-rapid-clicks?rq=1

Reply:
The button is enabled when the tree is saved and the MapFragment is navigated to. So, the user is able to add the tree again. I tested that.

I believe even if we introduce a delay, if the app was still processing the frames, it would cause duplicate onClick process to execute the navigate action
while on the NewFragment causing similar exception.

Reply:
Thanks, this looks good then. I'd just like to ask if you can reuse the view reference we already have for that button instead of finding it again, and moving it to the check for when only the addTreeButton is clicked.

Reply:
Thank you for the feedback. I updated the code (and tested the issue is resolved) as per your suggestions.